docs/rofl/app.mdx

```diff
@@ -36,220 +41,222 @@ chains).
-to specify a version to use by creating a special file called
-`rust-toolchain.toml` containing the following information:
-
-<!-- markdownlint-disable line-length -->
-![code toml](../../examples/runtime-sdk/rofl-oracle/rust-toolchain.toml "rust-toolchain.toml")
-<!-- markdownlint-enable line-length -->
+```
+Created ROFL application: rofl1qqn9xndja7e2pnxhttktmecvwzz0yqwxsquqyxdf
+```
 
-Additionally, due to the requirements of some upstream dependencies, you need to
-configure Cargo to always build with specific target CPU platform features
-(namely AES-NI and SSE3) by creating a `.cargo/config.toml` file with the
-following content:
+The app deployer account automatically becomes the initial admin of the app so
+it can update the app's configuration. The admin address can always be changed
+by the current admin.
 
 :::info
 
-You do not need this additional configuration if you're building with the
-[`rofl-dev`][rofl-dev] container, since that already has the relevant environment
-variables set appropriately.
+While the CLI implements a simple governance mechanism where the admin of the
+ROFL app is a single account, even a smart contract can be the admin. This
+allows for implementation of advanced agent governance mechanisms, like using
+multi-sigs or DAOs with veto powers to control the upgrade process.
 
 :::
 
-```toml title=".cargo/config.toml"
-[build]
-rustflags = ["-C", "target-feature=+aes,+ssse3"]
-rustdocflags = ["-C", "target-feature=+aes,+ssse3"]
+## Oracle Contract
 
-[test]
-rustflags = ["-C", "target-feature=+aes,+ssse3"]
-rustdocflags = ["-C", "target-feature=+aes,+ssse3"]
-```
+:::info
 
-After you complete this guide, the directory structure will look as follows:
+While we are using [EVM-based smart contracts] in this example, the on-chain
+part can be anything from a [WASM-based smart contract] to a dedicated
+[runtime module].
 
-```
-rofl-oracle
-├── .cargo
-│   └── config.toml      # Cargo configuration.
-├── Cargo.lock           # Rust dependency tree checksums.
-├── Cargo.toml           # Rust crate defintion.
-├── rust-toolchain.toml  # Rust toolchain version configuration.
-└── src
-    └── main.rs          # The ROFL app definition.
+[EVM-based smart contracts]: https://github.com/oasisprotocol/docs/blob/main/docs/build/sapphire/README.mdx
+[WASM-based smart contract]: https://github.com/oasisprotocol/docs/blob/main/docs/build/tools/other-paratimes/cipher/README.mdx
+[runtime module]: https://github.com/oasisprotocol/oasis-sdk/blob/main/docs/runtime/modules.md
+
+:::
+
+We have prepared a simple oracle contract for this example. You can find it by
+checking out the [prepared example project] from the Oasis SDK repository. It
+contains a simple [Oracle.sol] contract which collects observations from
+authenticated ROFL app instances, performs trivial aggregation and stores the
+final aggregated result. Read the [Sapphire quickstart] chapter to learn how to
```

Comment:
More a question for sapphire/ @matevz. We use `hardhat` in this guide, and I see that the "Sapphire quickstart" also uses `hardhat`. I thought `foundry` was the cool thing to use with Sapphire these days.

Reply:
This is copied over from the old ROFL tutorial. Not sure if anything has changed here.